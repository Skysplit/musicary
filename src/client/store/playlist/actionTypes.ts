export const FETCH_MANY_PLAYLISTS_REQUEST = 'playlist/FETCH_MANY_REQUEST';
export const FETCH_MANY_PLAYLISTS_SUCCESS = 'playlist/FETCH_MANY_SUCCESS';
export const FETCH_MANY_PLAYLISTS_FAILURE = 'playlist/FETCH_MANY_FAILURE';

export const FETCH_SINGLE_PLAYLIST_REQUEST = 'playlist/FETCH_SINGLE_REQUEST';
export const FETCH_SINGLE_PLAYLIST_SUCCESS = 'playlist/FETCH_SINGLE_SUCCESS';
export const FETCH_SINGLE_PLAYLIST_FAILURE = 'playlist/FETCH_SINGLE_FAILURE';

export const REMOVE_PLAYLIST_REQUEST = 'playlist/REMOVE_REQUEST';
export const REMOVE_PLAYLIST_SUCCESS = 'playlist/REMOVE_SUCCESS';
export const REMOVE_PLAYLIST_FAILURE = 'playlist/REMOVE_FAILURE';

export const SET_PLAYLIST = 'playlist/SET';
