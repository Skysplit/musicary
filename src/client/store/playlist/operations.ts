import { Dispatch } from 'react-redux';
import {
  fetchPlaylistsRequest,
  fetchPlaylistsSuccess,
  fetchPlaylistsFailure,
  fetchPlaylistRequest,
  fetchPlaylistSuccess,
  fetchPlaylistFailure,
  setSinglePlaylist,
  removePlaylistRequest,
  removePlaylistSuccess,
  removePlaylistFailure,
} from '@client/store/playlist/actions';
import client from '@client/utils/client';
import { PlaylistInterface } from '@server/module/playlist/playlist.model';

export const fetchManyPlaylists = () => async (dispatch: Dispatch) => {
  dispatch(fetchPlaylistsRequest());
  try {
    const response = await client.get('/playlists');
    dispatch(fetchPlaylistsSuccess(response.data));
  } catch (err) {
    dispatch(fetchPlaylistsFailure());
  }
};

export const fetchSinglePlaylist = (id: number) => async (dispatch: Dispatch) => {
  dispatch(fetchPlaylistRequest(id));
  try {
    const response = await client.get(`/playlists/${id}`);
    dispatch(fetchPlaylistSuccess(response.data));
  } catch (err) {
    dispatch(fetchPlaylistFailure(id));
  }
};

export const setPlaylist = (playlist: PlaylistInterface) => (dispatch: Dispatch) => {
  dispatch(setSinglePlaylist(playlist));
};

export const removePlaylist = (id: number) => async (dispatch: Dispatch) => {
  dispatch(removePlaylistRequest(id));
  try {
    await client.delete(`/playlists/${id}`);
    dispatch(removePlaylistSuccess(id));
  } catch (err) {
    dispatch(removePlaylistFailure(id));
  }
};
