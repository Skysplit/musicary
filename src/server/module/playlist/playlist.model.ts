import {
  BaseEntity,
  Entity,
  PrimaryGeneratedColumn,
  Column,
  ManyToMany,
  ManyToOne,
  JoinTable,
} from 'typeorm';
import { Length } from 'class-validator';
import User from '@server/module/user/user.model';
import Track from './track/track.model';

export interface PlaylistInterface {
  id?: number;
  name: string;
  position?: number;
  user?: User;
  tracks: Track[] | Promise<Track[]>;
}

@Entity()
export default class Playlist extends BaseEntity implements PlaylistInterface {
  @PrimaryGeneratedColumn()
  id: number;

  @Length(3, 150)
  @Column()
  name: string;

  @Column('tinyint')
  position: number;

  @ManyToOne(type => User, user => user.playlists)
  user: User;

  @ManyToMany(type => Track, track => track.playlists, { lazy: true })
  @JoinTable({ name: 'playlist_track' })
  tracks: Track[] | Promise<Track[]>;
}
