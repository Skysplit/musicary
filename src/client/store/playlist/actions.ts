import {
  FETCH_MANY_PLAYLISTS_REQUEST,
  FETCH_MANY_PLAYLISTS_SUCCESS,
  FETCH_SINGLE_PLAYLIST_REQUEST,
  FETCH_MANY_PLAYLISTS_FAILURE,
  FETCH_SINGLE_PLAYLIST_FAILURE,
  FETCH_SINGLE_PLAYLIST_SUCCESS,
  SET_PLAYLIST,
  REMOVE_PLAYLIST_REQUEST,
  REMOVE_PLAYLIST_FAILURE,
  REMOVE_PLAYLIST_SUCCESS,
  REMOVE_PLAYLIST_TRACK,
} from '@client/store/playlist/actionTypes';
import { PlaylistInterface } from '@client/store/playlist';

// Fetching many playlists
export const fetchPlaylistsRequest = () => ({
  type: FETCH_MANY_PLAYLISTS_REQUEST as typeof FETCH_MANY_PLAYLISTS_REQUEST,
});

export const fetchPlaylistsSuccess = (payload: PlaylistInterface[]) => ({
  payload,
  type: FETCH_MANY_PLAYLISTS_SUCCESS as typeof FETCH_MANY_PLAYLISTS_SUCCESS,
});

export const fetchPlaylistsFailure = () => ({
  type: FETCH_MANY_PLAYLISTS_FAILURE as typeof FETCH_MANY_PLAYLISTS_FAILURE,
});

// Fetching single playlist
export const fetchPlaylistRequest = (id: number) => ({
  payload: id,
  type: FETCH_SINGLE_PLAYLIST_REQUEST as typeof FETCH_SINGLE_PLAYLIST_REQUEST,
});

export const fetchPlaylistSuccess = (payload: PlaylistInterface) => ({
  payload,
  type: FETCH_SINGLE_PLAYLIST_SUCCESS as typeof FETCH_SINGLE_PLAYLIST_SUCCESS,
});

export const fetchPlaylistFailure = (id: number) => ({
  payload: id,
  type: FETCH_SINGLE_PLAYLIST_FAILURE as typeof FETCH_SINGLE_PLAYLIST_FAILURE,
});

// Delete single playlist
export const removePlaylistRequest = (id: number) => ({
  payload: id,
  type: REMOVE_PLAYLIST_REQUEST as typeof REMOVE_PLAYLIST_REQUEST,
});

export const removePlaylistSuccess = (id: number) => ({
  payload: id,
  type: REMOVE_PLAYLIST_SUCCESS as typeof REMOVE_PLAYLIST_SUCCESS,
});

export const removePlaylistFailure = (id: number) => ({
  payload: id,
  type: REMOVE_PLAYLIST_FAILURE as typeof REMOVE_PLAYLIST_FAILURE,
});

// Save single playlist
export const setSinglePlaylist = (payload: PlaylistInterface) => ({
  payload,
  type: SET_PLAYLIST as typeof SET_PLAYLIST,
});

export const removePlaylistTrack = (playlistId: number, trackId: number) => ({
  type: REMOVE_PLAYLIST_TRACK as typeof REMOVE_PLAYLIST_TRACK,
  payload: {
    playlistId,
    trackId,
  },
});

