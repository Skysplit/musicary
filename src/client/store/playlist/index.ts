import { omit, map, uniq, reject } from 'lodash';
import {
  fetchPlaylistRequest,
  fetchPlaylistSuccess,
  fetchPlaylistsRequest,
  fetchPlaylistsSuccess,
  fetchPlaylistFailure,
  fetchPlaylistsFailure,
  setSinglePlaylist,
  removePlaylistRequest,
  removePlaylistSuccess,
  removePlaylistFailure,
  removePlaylistTrack,
} from '@client/store/playlist/actions';
import {
  FETCH_SINGLE_PLAYLIST_REQUEST,
  FETCH_SINGLE_PLAYLIST_SUCCESS,
  FETCH_MANY_PLAYLISTS_REQUEST,
  FETCH_MANY_PLAYLISTS_SUCCESS,
  FETCH_MANY_PLAYLISTS_FAILURE,
  FETCH_SINGLE_PLAYLIST_FAILURE,
  SET_PLAYLIST,
  REMOVE_PLAYLIST_REQUEST,
  REMOVE_PLAYLIST_SUCCESS,
  REMOVE_PLAYLIST_FAILURE,
  REMOVE_PLAYLIST_TRACK,
} from '@client/store/playlist/actionTypes';
import { fetchTracksSuccess, setSingleTrack } from '@client/store/track/actions';
import { FETCH_MANY_TRACKS_SUCCESS, SET_TRACK } from '@client/store/track/actionTypes';

export interface PlaylistInterface {
  id?: number;
  name: string;
  position?: number;
  tracks?: number[];
}

export interface PlaylistState {
  isLoading: boolean;
  byId: {
    [key: number]: {
      isLoading: boolean;
      isRemoving: boolean;
      data?: PlaylistInterface;
    };
  };
}

type Actions =
  | ReturnType<typeof fetchPlaylistRequest>
  | ReturnType<typeof fetchPlaylistSuccess>
  | ReturnType<typeof fetchPlaylistFailure>
  | ReturnType<typeof fetchPlaylistsRequest>
  | ReturnType<typeof fetchPlaylistsSuccess>
  | ReturnType<typeof fetchPlaylistsFailure>
  | ReturnType<typeof setSinglePlaylist>
  | ReturnType<typeof removePlaylistRequest>
  | ReturnType<typeof removePlaylistSuccess>
  | ReturnType<typeof removePlaylistFailure>
  | ReturnType<typeof fetchTracksSuccess>
  | ReturnType<typeof setSingleTrack>
  | ReturnType<typeof removePlaylistTrack>;


const initialState: PlaylistState = {
  isLoading: false,
  byId: {},
};

export default (state: PlaylistState = initialState, action: Actions): PlaylistState => {
  switch (action.type) {
    case FETCH_MANY_PLAYLISTS_REQUEST: {
      return {
        ...state,
        isLoading: true,
      };
    }

    case FETCH_MANY_PLAYLISTS_SUCCESS: {
      return {
        ...state,
        isLoading: false,
        byId: action.payload.reduce(
          (accum, data) => Object.assign(accum, {
            [data.id]: { data, isLoading: false },
          }),
          {},
        ),
      };
    }

    case FETCH_MANY_PLAYLISTS_FAILURE: {
      return state;
    }

    case FETCH_SINGLE_PLAYLIST_REQUEST: {
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload]: {
            isLoading: true,
            isRemoving: false,
          },
        },
      };
    }

    case FETCH_SINGLE_PLAYLIST_SUCCESS: {
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload.id]: {
            isLoading: false,
            isRemoving: false,
            data: action.payload,
          },
        },
      };
    }

    case FETCH_SINGLE_PLAYLIST_FAILURE: {
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload]: {
            isLoading: false,
            isRemoving: false,
          },
        },
      };
    }

    case REMOVE_PLAYLIST_REQUEST: {
      const item = state.byId[action.payload];
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload]: {
            ...item,
            isRemoving: true,
          },
        },
      };
    }

    case REMOVE_PLAYLIST_SUCCESS: {
      return {
        ...state,
        byId: omit(state.byId, [action.payload]),
      };
    }

    case REMOVE_PLAYLIST_FAILURE: {
      const item = state.byId[action.payload];
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload]: {
            ...item,
            isRemoving: false,
          },
        },
      };
    }

    case SET_PLAYLIST: {
      return {
        ...state,
        byId: {
          ...state.byId,
          [action.payload.id]: {
            isLoading: false,
            isRemoving: false,
            data: action.payload,

          },
        },
      };
    }

    case FETCH_MANY_TRACKS_SUCCESS: {
      const { playlistId, tracks } = action.payload;
      const playlist = state.byId[playlistId];

      if (!playlist) {
        return state;
      }

      return {
        ...state,
        byId: {
          ...state.byId,
          [playlistId]: {
            ...playlist,
            data: {
              ...playlist.data,
              tracks: map(tracks, 'id'),
            },
          },
        },
      };
    }

    case SET_TRACK: {
      const { playlistId, track } = action.payload;
      const playlist = state.byId[playlistId];

      if (!playlist) {
        return;
      }

      return {
        ...state,
        byId: {
          ...state.byId,
          [playlistId]: {
            ...playlist,
            data: {
              ...playlist.data,
              tracks: uniq([
                ...playlist.data.tracks,
                track.id,
              ]),
            },
          },
        },
      };
    }

    case REMOVE_PLAYLIST_TRACK: {
      const { playlistId, trackId } = action.payload;
      const playlist = state.byId[playlistId];

      if (!playlist) {
        return;
      }

      return {
        ...state,
        byId: {
          ...state.byId,
          [playlistId]: {
            ...playlist,
            data: {
              ...playlist.data,
              tracks: reject(playlist.data.tracks, id => id === trackId),
            },
          },
        },
      };
    }

    default: {
      return state;
    }
  }
};
