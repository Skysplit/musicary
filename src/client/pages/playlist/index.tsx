import React, { PureComponent } from 'react';
import { Request } from 'express';
import { bindActionCreators } from 'redux';
import { connect, Dispatch } from 'react-redux';
import Error from 'next/error';
import { Context } from 'next/document';
import PlaylistsListContainer from '@client/containers/PlaylistsListContainer';
import client, { getHeaders } from '@client/utils/client';
import { PlaylistInterface } from '@client/store/playlist';
import { fetchPlaylistsSuccess } from '@client/store/playlist/actions';

interface PropsContext extends Context {
  req: Request;
}

interface InitialProps {
  title: string;
  errorCode?: number;
  playlists?: PlaylistInterface[];
}

interface PlaylistsPageProps extends InitialProps {
  fetchPlaylistsSuccess: typeof fetchPlaylistsSuccess;
}

export class PlaylistsPage extends PureComponent<PlaylistsPageProps> {
  static async fetchPlaylists(req: Request) {
    const headers = getHeaders(req);
    const response = await client.get('/playlists', { headers });
    return response.data as PlaylistInterface[];
  }

  static async getInitialProps(ctx: PropsContext): Promise<InitialProps> {
    let errorCode: number;
    let playlists: PlaylistInterface[];

    try {
      playlists = await PlaylistsPage.fetchPlaylists(ctx.req);
    } catch (err) {
      errorCode = err.response.status;
    }

    return {
      errorCode,
      playlists,
      title: 'My playlists',
    };
  }

  constructor(props: PlaylistsPageProps) {
    super(props);

    if (props.playlists) {
      props.fetchPlaylistsSuccess(props.playlists);
    }
  }

  render() {
    const { errorCode } = this.props;

    if (errorCode) {
      return <Error statusCode={errorCode} />;
    }

    return (
      <PlaylistsListContainer />
    );
  }
}

const actionCreators = {
  fetchPlaylistsSuccess,
};

const mapDispatchToProps = (dispatch: Dispatch) => bindActionCreators(actionCreators, dispatch);

export default connect(null, mapDispatchToProps)(PlaylistsPage);
