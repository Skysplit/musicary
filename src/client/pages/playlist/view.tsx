import { PureComponent, Fragment } from 'react';
import { bindActionCreators } from 'redux';
import { connect, Dispatch } from 'react-redux';
import { Request } from 'express';
import io from 'socket.io-client';
import { Snackbar } from 'react-md';
import { Context } from 'next/document';
import Error from 'next/error';
import { PlaylistInterface } from '@client/store/playlist';
import { setSinglePlaylist, removePlaylistTrack } from '@client/store/playlist/actions';
import client, { getHeaders } from '@client/utils/client';
import { fetchTracksSuccess, setSingleTrack, removeTrack } from '@client/store/track/actions';
import { TrackInterface } from '@client/store/track';
import { getUserToken } from '@client/utils/userData';
import PlaylistViewContainer from '@client/containers/PlaylistViewContainer';

interface PageContext extends Context {
  req: Request;
}

interface InitialProps {
  title: string;
  playlist?: PlaylistInterface;
  tracks?: TrackInterface[];
  errorCode?: number;
}

interface PlaylistViewPageProps extends InitialProps {
  setSinglePlaylist: typeof setSinglePlaylist;
  fetchTracksSuccess: typeof fetchTracksSuccess;
  setSingleTrack: typeof setSingleTrack;
  removePlaylistTrack: typeof removePlaylistTrack;
}

type ComponentProps = PlaylistViewPageProps & {};

interface Toast {
  text: string;
}

type ComponentState = {
  toasts: Toast[],
};

type AuthHeaders = ReturnType<typeof getHeaders>;

export class PlaylistViewPage extends PureComponent<ComponentProps, ComponentState> {
  private io: SocketIOClient.Socket;

  static async fetchPlaylist(id: number, headers: AuthHeaders) {
    const response = await client.get(`/playlists/${id}`, { headers });
    return response.data as PlaylistInterface;
  }

  static async fetchPlaylistTracks(id: number, headers: AuthHeaders) {
    const response = await client.get(`/playlists/${id}/tracks`, { headers });
    return response.data as TrackInterface[];
  }

  static async getInitialProps({ req, query }: PageContext): Promise<InitialProps> {
    let playlist: PlaylistInterface;
    let tracks: TrackInterface[];
    let errorCode: number;
    let title = 'Playlist';

    try {
      const headers = getHeaders(req);

      [playlist, tracks] = await Promise.all([
        PlaylistViewPage.fetchPlaylist(query.id as number, headers),
        PlaylistViewPage.fetchPlaylistTracks(query.id as number, headers),
      ]);

      title = `Playlist | ${playlist.name}`;
    } catch (err) {
      errorCode = err.response.status;
    }

    return {
      title,
      playlist,
      tracks,
      errorCode,
    };
  }

  state: ComponentState = {
    toasts: [],
  };

  constructor(props: ComponentProps) {
    super(props);

    if (props.playlist) {
      props.setSinglePlaylist(props.playlist);
      props.fetchTracksSuccess(props.playlist.id, props.tracks);
    }
  }

  private handleSocketConnection = () => {
    const { toasts } = this.state;
    const toast = { text: 'Connected with server!' };

    this.setState({
      toasts: [...toasts, toast],
    });
  }

  private handleSocketImportError = (playlistId: number) => {
    const { playlist } = this.props;
    const { toasts } = this.state;

    if (playlist.id !== playlistId) {
      return;
    }

    const toast = {
      text: 'There was an error while importing playlist',
    };

    this.setState({
      toasts: [...toasts, toast],
    });
  }

  private handleSocketTrack = (playlistId: number, track: TrackInterface) => {
    const { playlist } = this.props;
    const { toasts } = this.state;

    if (playlist.id !== playlistId) {
      return;
    }

    const toast = {
      text: `New track! ${track.name}`,
    };

    this.props.setSingleTrack(playlistId, track);

    this.setState({
      toasts: [...toasts, toast],
    });
  }

  private handleSocketTracksAccepted = (playlistId: number) => {
    const { playlist } = this.props;
    const { toasts } = this.state;

    if (playlist.id !== playlistId) {
      return;
    }

    const toast = {
      text: `Your tracks are being processed. They will appear on your list soon`,
    };

    this.setState({
      toasts: [...toasts, toast],
    });
  }

  private handleSocketTrackRemoved = (playlistId: number, track: TrackInterface) => {
    const { playlist } = this.props;
    const { toasts } = this.state;

    if (playlist.id !== playlistId) {
      return;
    }

    const toast = {
      text: `Track ${track.name} removed from playlist`,
    };

    this.setState({
      toasts: [...toasts, toast],
    });

    this.props.removePlaylistTrack(playlistId, track.id);
  }

  componentDidMount() {
    this.io = io({
      query: { token: getUserToken() },
      autoConnect: false,
    });

    this.io.connect();

    this.io.on('connect', this.handleSocketConnection);
    this.io.on('tracks-accepted', this.handleSocketTracksAccepted);
    this.io.on('track', this.handleSocketTrack);
    this.io.on('import-error', this.handleSocketImportError);
    this.io.on('track-removed', this.handleSocketTrackRemoved);
  }

  componentWillUnmount() {
    this.io.removeAllListeners();
    this.io.close();
  }

  handleDismiss = () => {
    const [, ...toasts] = this.state.toasts;
    this.setState({ toasts });
  }

  render() {
    const { errorCode, playlist } = this.props;

    if (errorCode) {
      return <Error statusCode={errorCode} />;
    }

    return (
      <Fragment>
        <PlaylistViewContainer playlistId={playlist.id} />
        <Snackbar
          toasts={this.state.toasts}
          onDismiss={this.handleDismiss}
          autohideTimeout={2000}
          autohide
        />
      </Fragment>
    );
  }
}

const actionCreators = {
  setSinglePlaylist,
  setSingleTrack,
  removePlaylistTrack,
  fetchTracksSuccess,
};

const mapDispatchToProps = (dispatch: Dispatch) => bindActionCreators(actionCreators, dispatch);

export default connect(null, mapDispatchToProps)(PlaylistViewPage);
