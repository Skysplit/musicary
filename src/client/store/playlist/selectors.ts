import { filter, map, sortBy, flow } from 'lodash/fp';
import { createSelector } from 'reselect';
import { State } from '@client/store';
import { PlaylistInterface } from '@server/module/playlist/playlist.model';

interface Props {
  id?: number;
}

const playlistsById = (state: State) => state.playlist.byId;

export const getPlaylists = createSelector(
  [playlistsById],
  (playlists) => {
    const loadedPlaylists: PlaylistInterface[] = flow([
      filter({ isLoading: false }),
      map('data'),
      sortBy('position'),
    ])(playlists);

    return loadedPlaylists;
  },
);

export const getIsPlaylistRemoving = (state: State, props: Props) => (
  getPlaylist(state, props).isRemoving
);

export const getPlaylist = (state: State, props: Props) => state.playlist.byId[props.id];
