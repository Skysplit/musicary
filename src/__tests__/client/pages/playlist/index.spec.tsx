import React from 'react';
import { PlaylistsPage } from '@client/pages/playlist/index';
import { shallow } from 'enzyme';

describe('pages/home', () => {
  const defaultProps = {
    fetchPlaylistsSuccess: () => {},
  };

  test('should render component', () => {
    const wrapper = shallow(<PlaylistsPage {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render error page', () => {
    const props = {
      errorCode: 401,
    };

    const wrapper = shallow(<PlaylistsPage {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should send action when playlist are fetched', () => {
    const props = {
      fetchPlaylistsSuccess: jest.fn(),
      playlists: ['playlist1', 'playlist2'],
    };

    const wrapper = shallow(<PlaylistsPage {...props} />);

    expect(props.fetchPlaylistsSuccess).toHaveBeenCalledWith(props.playlists);
  });
});
