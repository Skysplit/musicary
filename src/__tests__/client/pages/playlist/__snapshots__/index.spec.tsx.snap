// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/home should render component 1`] = `<Connect(PlaylistsListContainer) />`;

exports[`pages/home should render error page 1`] = `
<Error
  statusCode={401}
/>
`;
